cinema/tests/test_movie_session_api.py

```diff
@@ -0,0 +1,172 @@
+        self.assertIn(serializer_movie_default_title.data, res.data)
+        self.assertNotIn(serializer_movie_with_test_title.data, res.data)
+
+    def test_filter_movie_by_actors(self):
+        movie_without_actors = sample_movie(
+            title="test title",
+            description="test description",
+            duration=100
+        )
+        actor_1 = Actor.objects.create(first_name="Bill", last_name="Murray")
+        actor_2 = Actor.objects.create(first_name="Andie", last_name="MacDowell")
+        movie_with_actors = sample_movie()
+        movie_with_actors.actors.add(
+            actor_1,
+            actor_2,
+        )
+        res = self.client.get(MOVIE_URL, data={"actors": f"{actor_1.id},{actor_2.id}"})
+        serializer_movie_with_actors = MovieListSerializer(movie_with_actors)
+        serializer_movie_without_actors = MovieListSerializer(movie_without_actors)
+        self.assertIn(serializer_movie_with_actors.data, res.data)
+        self.assertNotIn(serializer_movie_without_actors.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+        serializer = MovieDetailSerializer(res.data)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def create_movie_forbidden(self):
+        payload = {
+            "title": "Test title",
+            "description": "Test description",
+            "duration": 123,
+        }
+        res = self.client.post(path=MOVIE_URL, data=payload)
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="admin@admin.com",
+            password="testpassword",
+            is_staff=True,
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_create_movie(self):
+        Genre.objects.create(name="Comedy")
+        payload = {
+            "title": "Test title",
+            "description": "Test description",
+            "duration": 123,
+        }
+        res = self.client.post(path=MOVIE_URL, data=payload)
+        movie = Movie.objects.get(id=res.data["id"])
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        for key in payload:
+            self.assertEqual(payload[key], getattr(movie, key))
+
+    def test_create_movie_with_genres(self):
+        genre_1 = Genre.objects.create(name="Comedy")
+        genre_2 = Genre.objects.create(name="Drama")
+        genre_3 = Genre.objects.create(name="Fantasy")
+        genre_4 = Genre.objects.create(name="Romance")
+        payload = {
+            "title": "Test title",
+            "description": "Test description",
+            "duration": 123,
+            "genres": 1
```

Comment:
The payload for genres should be a list of IDs, not a single integer. Change `'genres': 1` to `'genres': [genre_1.id, genre_2.id, genre_3.id, genre_4.id]`.